bc_obps/compliance/service/automated_process/compliance_handlers.py

```diff
@@ -0,0 +1,110 @@
+        if obligation.penalty_status != ComplianceObligation.PenaltyStatus.PAID:
+            ComplianceObligationService.update_penalty_status(obligation.pk, ComplianceObligation.PenaltyStatus.PAID)
+            logger.info(f"Updated penalty status to PAID for obligation {obligation.obligation_id}")
+
+
+class PenaltyAccruingHandler(ComplianceUpdateHandler):
+    """Handler for handling obligations that need to start accruing penalties."""
+
+    def can_handle(self, invoice: ElicensingInvoice, obligation: ComplianceObligation) -> bool:
+        """Check if obligation should start accruing penalties."""
+        has_penalty = hasattr(invoice, 'compliance_penalty') and getattr(invoice, 'compliance_penalty', None)
+        if has_penalty:
+            return False
+
+        return (
+            obligation.compliance_report_version.status == ComplianceReportVersion.ComplianceStatus.OBLIGATION_NOT_MET
+            and invoice.outstanding_balance > Decimal('0.00')
+            and invoice.due_date < timezone.now()
+        )
+
+    def handle(self, invoice: ElicensingInvoice, obligation: ComplianceObligation) -> None:
+        """Update penalty status to ACCRUING and create penalty if needed."""
+        if obligation.penalty_status != ComplianceObligation.PenaltyStatus.ACCRUING:
+            ComplianceObligationService.update_penalty_status(
+                obligation.pk, ComplianceObligation.PenaltyStatus.ACCRUING
+            )
+            logger.info(f"Updated penalty status to ACCRUING for obligation {obligation.obligation_id}")
+
+        if not getattr(obligation, 'compliance_penalty', None):
+            PenaltyCalculationService.create_penalty(obligation)
+            logger.info(f"Created penalty for obligation {obligation.obligation_id}")
+        else:
+            logger.info(f"Penalty already exists for obligation {obligation.obligation_id}")
+
+
+class ObligationPaidHandler(ComplianceUpdateHandler):
+    """Handler for handling obligations that are fully paid."""
+
+    def can_handle(self, invoice: ElicensingInvoice, obligation: ComplianceObligation) -> bool:
+        """Check if obligation is fully paid and should be updated to NO_OBLIGATION_OR_EARNED_CREDITS."""
+        return (
+            obligation.compliance_report_version.status == ComplianceReportVersion.ComplianceStatus.OBLIGATION_NOT_MET
+            and invoice.outstanding_balance == Decimal('0.00')
+        )
+
+    def handle(self, invoice: ElicensingInvoice, obligation: ComplianceObligation) -> None:
+        """Update compliance status to NO_OBLIGATION_OR_EARNED_CREDITS."""
+        compliance_report_version = obligation.compliance_report_version
+        compliance_report_version.status = ComplianceReportVersion.ComplianceStatus.NO_OBLIGATION_OR_EARNED_CREDITS
+        compliance_report_version.save(update_fields=['status'])
+        logger.info(f"Updated compliance status for obligation {obligation.obligation_id}")
+
```

Comment:
This is the point where we should create a penalty if the obligation was paid to 0 after the deadline. We can add the following here to handle that
```suggestion
if invoice.due_date < timezone.now():
            PenaltyCalculationService.create_penalty(obligation)
            logger.info(f"Created penalty for obligation {obligation.obligation_id}")
```